### PR TITLE
lambda support vpc

### PR DESCRIPTION
**Pull request by gabbyhagag**
Support VPC to lambda.
After this change, you can add VPC configuration to lambda and it will add the policy.
```
{
  vpc_config = {
    subnet_ids         = ["subnet-12345678", "subnet-87654321"]
    security_group_ids = ["sg-12345678"]
  }
}
```

### CODE REVIEW

File: examples/basic-terraform-integration.tf [hunk 1]
```diff
@@ -7,4 +7,10 @@ module "spectral_lambda_integration" {
     SPECTRAL_DSN = "MySpectralDSN"
     CHECK_POLICY = "Fail on any issue" # (Fail on any issue / Fail on warnings and above / Fail on errors only / Always Pass)
   }
+  
+  # With VPC configuration
```
Comment (ChengaDev) [inline examples/basic-terraform-integration.tf]: Not sure if this basic configuration, I'd create a new dedicated example file for setting up VPC

File: modules/lambda/lambda.tf [hunk 2]
```diff
@@ -21,6 +23,18 @@ resource "aws_lambda_function" "spectral_scanner_lambda" {
   environment {
     variables = var.env_vars
   }
+
+  dynamic "vpc_config" {
```
Comment (ChengaDev) [inline modules/lambda/lambda.tf:27]: Why is it dynamic with foreach and not count?

File: examples/basic-terraform-integration.tf (hunk 1 above)
Comment (gabbyhagag) [reply examples/basic-terraform-integration.tf]: fixed

File: modules/lambda/lambda.tf (hunk 2 above)
Comment (gabbyhagag) [reply modules/lambda/lambda.tf:27]: 1. for_each with dynamic blocks can be more readable, especially when using nested configurations.
2. The vpc_config block on "aws_lambda_function" resource does not support "count" and that's why to use dynamic, for example, to use count, I need to make a duplicate "lambda" resource with the vpc_config and the ones without.

**Review by ChengaDev — APPROVED** [review]

File: examples/lambda-in-vpc.tf
Comment (eladbash) [inline examples/lambda-in-vpc.tf:1]: vpc in lambda

File: examples/lambda-in-vpc.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module "spectral_lambda_integration" {
+  source = "github.com/SpectralOps/spectral-terraform-lambda-integration"
+
+  integration_type = "gitlab"
+
+  env_vars = {
+    # Required environment variables 
+    SPECTRAL_DSN          = "MySpectralDSN"
```
Comment (eladbash) [inline examples/lambda-in-vpc.tf]: I think better avoid spectral dsn in the examples - the best practice is to use vault since this is a secret

Comment (chkp-gavrielh) [reply examples/lambda-in-vpc.tf]: Fixed:
https://github.com/SpectralOps/spectral-terraform-lambda-integration/pull/20/commits/9007bf3ed2f04aa25b776a9c82f2d4bfb7fc9fb7

File: examples/lambda-in-vpc.tf
Comment (chkp-gavrielh) [reply examples/lambda-in-vpc.tf:1]: Fixed:
https://github.com/SpectralOps/spectral-terraform-lambda-integration/pull/20/commits/9007bf3ed2f04aa25b776a9c82f2d4bfb7fc9fb7

**Review by eladbash — APPROVED** [review]

**Pull request merged**